backend/services/session_manager.py

```
@@ -0,0 +1,595 @@
+    """Manages agent sessions, memory, and caching"""
+    
+    def __init__(self):
+        self.active_sessions: Dict[str, Dict[str, Any]] = {}
+        self.session_timeout = settings.session_timeout
+        self.cache_ttl = settings.cache_ttl
+        
+    def generate_session_id(self, user_id: int, scenario_id: int, scene_id: int) -> str:
+        """Generate unique session ID"""
+        timestamp = datetime.utcnow().isoformat()
+        session_data = f"{user_id}_{scenario_id}_{scene_id}_{timestamp}"
+        return hashlib.md5(session_data.encode()).hexdigest()
```

Ruff (0.12.2)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)